http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/RuntimeServiceInterp.scala

```diff
@@ -530,6 +532,53 @@ class RuntimeServiceInterp[F[_]: Parallel](config: RuntimeServiceConfig,
                                            Some(traceId))
         publisherQueue.enqueue1(message).as(message)
       }
+      // we only need to return the message that might cause a start/stop transition
+    } yield msg
+
+  private[service] def processUpdateGceWithPdConfigRequest(req: UpdateRuntimeConfigRequest.GceConfig,
+                                                           allowStop: Boolean,
+                                                           runtime: Runtime,
+                                                           gceConfig: RuntimeConfig.GceWithPdConfig,
+                                                           traceId: TraceId): F[Option[UpdateRuntimeMessage]] =
+    for {
+      // should machine type be updated?
+      targetMachineType <- traverseIfChanged(req.updatedMachineType, gceConfig.machineType) { mt =>
```

Comment:
can you make this into a function so that it can be reused in `processUpdateGceConfigRequest`